native/cocos/core/scene-graph/Node.h

```diff
@@ -474,19 +473,19 @@ class Node : public BaseNode {
 
     inline Vec3 getForward() const {
         Vec3 forward{0, 0, -1};
-        forward.transformQuat(_worldRotation);
+        forward.transformQuat(getWorldRotation());
```

Comment:
Critical bug: If world transform is dirty, _worldRotation will be wrong value.
getWorldRotation internal will updateWorldTransform if it's needed.
```c++
const Quaternion &Node::getWorldRotation() const { //NOLINT(misc-no-recursion)
    const_cast<Node *>(this)->updateWorldTransform();
    return _worldRotation;
}
```